src/Microsoft.Azure.WebJobs.Extensions.Kafka/Trigger/SingleItemFunctionExecutor.cs

```diff
@@ -72,7 +76,17 @@ private async Task ProcessPartitionItemsAsync(int partition, IEnumerable<IKafkaE
                     TriggerValue = triggerInput,
                 };
 
-                await this.ExecuteFunctionAsync(triggerData, cancellationToken);
+                using (var singleEventActivityProvider = new SingleEventActivityProvider(kafkaEventData, consumerGroup)) { 
```

Comment:
This is a complete behavior of execution and adding activity & setting status so should be a separate method

Reply:
Added new method for setting status